src/js/apps/patients/patient/patient_app.js

```diff
@@ -61,39 +67,33 @@ export default SubRouterApp.extend({
     this.showView();
   },
 
-  startPatientAction(patientId, actionId) {
-    const actionApp = this.getChildApp('action');
+  startList(list) {
+    if (!this.isRunning() || this._list === list) return;
 
-    this.listenToOnce(actionApp, {
-      'start'(options, action) {
-        this.editActionList(action);
-      },
-      'fail'() {
-        this.startCurrent('dashboard');
-      },
-    });
+    this._list = list;
 
-    this.startChildApp('action', { actionId, patientId });
+    this.startCurrent(list);
   },
 
-  startActionList(action) {
-    if (action.isDone()) return this.startCurrent('archive');
+  startPatientAction(list, patientId, actionId) {
+    if (!this.isRunning()) {
+      this.actionId = actionId;
```

Comment:
this is awkward.. this is the `onBeforeStart` handling the route.. if we change this to the parent router passing the actionId so that it is available easily to the `beforeStart` then what we end up with is the list reloading every time..  I think this is potentially an issue with the subrouter itself.. and I'm not sure if it's better to fix that now.. or to reinforce this pattern